test/date.js

```diff
@@ -32,6 +32,16 @@ describe('date', () => {
         ], done);
     });
 
+    it('fails on non-finite numbers', (done) => {
+
+        const schema = Joi.date();
+        Helper.validate(schema, [
+            [Infinity, false],
```

Comment:
From now on I'm trying to always test error messages as well (https://github.com/hapijs/joi/blob/master/test/any.js#L1463).